Sources/SwiftCliCore/InputValidator.swift

```diff
@@ -11,140 +24,175 @@ public class InputValidator {
+            let prevRow = allRowsMetaData[rowNum-1]
+            var currentRow = allRowsMetaData[rowNum]
+            if !isCurrentRowStitchCountValid(prevRow: prevRow, currentRow: currentRow) {
+                let expectedNextRowWidth = prevRow.width + currentRow.leftIncDec + currentRow.rightIncDec
+                return .failure(InputError.invalidRowWidth(
+                    invalidRowNumber: currentRow.userRowNumber,
+                    expectedStitchCount: expectedNextRowWidth,
+                    actualCount: currentRow.width
+                ))
+            }
+        }
+        return .success(allRowsMetaData)
     }
-    return .success(row)
-}
 
-private func validateEachRowWidth(allRowsMetaData: [RowInfo]) -> Result<[RowInfo], InputError> {
-    let numberOfRowsToCheck = (allRowsMetaData.count) - 1
-    if numberOfRowsToCheck == 0 {
-        return .success(allRowsMetaData)
+    private func isCurrentRowStitchCountValid(prevRow: RowInfo, currentRow: RowInfo) -> Bool {
+        let expectedCurrentRowWidth = prevRow.width + currentRow.leftIncDec + currentRow.rightIncDec
+        return expectedCurrentRowWidth == currentRow.width
     }
-    for rowNum in 1...numberOfRowsToCheck {
-        let prevRow = allRowsMetaData[rowNum-1]
-        var currentRow = allRowsMetaData[rowNum]
-        if !isCurrentRowStitchCountValid(prevRow: prevRow, currentRow: currentRow) {
-            let expectedNextRowWidth = prevRow.width + currentRow.leftIncDec + currentRow.rightIncDec
-            return .failure(InputError.invalidRowWidth(
-                invalidRowNumber: currentRow.userRowNumber,
-                expectedStitchCount: expectedNextRowWidth,
-                actualCount: currentRow.width
-            ))
+
+    public func knitFlatArray(array: [[String]]) -> [[String]] {
+        let numberofRows = array.count
+        var flatArray = array
+        for rowNum in 0..<numberofRows where rowNum % 2 == 1 {
+            flatArray[rowNum].reverse()
         }
+        return flatArray
     }
-    return .success(allRowsMetaData)
-}
-
-private func isCurrentRowStitchCountValid(prevRow: RowInfo, currentRow: RowInfo) -> Bool {
-    let expectedCurrentRowWidth = prevRow.width + currentRow.leftIncDec + currentRow.rightIncDec
-    return expectedCurrentRowWidth == currentRow.width
-}
 
-public func knitFlatArray(array: [[String]]) -> [[String]] {
-    let numberofRows = array.count
-    var flatArray = array
-    for rowNum in 0..<numberofRows where rowNum % 2 == 1 {
-        flatArray[rowNum].reverse()
+    private func checkRepeats(pattern: [[String]]) -> Result<Success, InputError> {
+
+        for (rowIndex, row) in pattern.enumerated() {
+            for (stitchIndex, stitch) in row.enumerated() {
+                if let _ = (stitch.range(of: "^[(0-9x)]*$", options: .regularExpression)) {
+                    let numberOfRepeats = Int(stitch.components(separatedBy: CharacterSet.decimalDigits.inverted).joined())
```

Comment:
We talked before about refactoring this regex, I'd like to see that before this project finishes.